### PR TITLE
Add ERC20 preset

### PR DESCRIPTION
**Pull request by andrew-fleming**
Fixes (partial) #804.

#### PR Checklist

- [x] Tests
- [x] Tried the feature on a public network (deploy [tx](https://testnet.starkscan.co/tx/0xe40ea1c06d155ba98d5b5122354627121c9ed1777c7a3453b89eb885443346#overview))
- [x] Documentation

### CODE REVIEW

File: docs/modules/ROOT/pages/erc20.adoc [hunk 1]
```diff
@@ -241,3 +241,13 @@ mod MyToken {
 This contract expects a `decimals` argument in the constructor and uses an internal function to write the decimals to storage.
 Note that the `decimals` state variable must be defined in the contract's storage because this variable does not exist in the component offered by OpenZeppelin Contracts for Cairo.
 It's important to include a custom ERC20 metadata implementation and NOT use the Contracts for Cairo `ERC20MetadataImpl` in this specific case since the `decimals` method will always return `18`.
+
+== Preset ERC20 contract
```
Comment (ericnordelo) [inline docs/modules/ROOT/pages/erc20.adoc:245]: https://github.com/OpenZeppelin/cairo-contracts/pull/803#issuecomment-1798461724

As mentioned in the comment, I think we should keep this in the references side of the documentation, and not in the understanding section.

File: docs/modules/ROOT/pages/guides/erc20-supply.adoc [hunk 2]
```diff
@@ -42,20 +42,20 @@ mod MyToken {
...
         self.erc20.initializer(name, symbol);
-        self.erc20._mint(recipient, initial_supply);
+        self.erc20._mint(recipient, fixed_supply);
     }
 }
 ----
 
 In the constructor, we're first calling the ERC20 initializer to set the token name and symbol.
-Next, we're calling the internal `_mint` function which creates `initial_supply` of tokens and allocates them to `recipient`.
+Next, we're calling the internal `_mint` function which creates `fixed_supply` of tokens and allocates them to `recipient`.
```
Comment (ericnordelo) [inline docs/modules/ROOT/pages/guides/erc20-supply.adoc:58]: 👍🏼

**Review by ericnordelo — COMMENTED** [review]
Looking great Andrew! After settling on how to approach documenting these presets, I think is good to go.

**Review by ericnordelo — APPROVED** [review]
Looking good! We still need to add the class_hash table somehow in the docs, but I think we can merge this PR and tackle that in another one, because that one is going to take some time no design and back and forth, and having these PRs staled is not necessary imo.

**Comment by andrew-fleming** [comment]
> Looking good! We still need to add the class_hash table somehow in the docs, but I think we can merge this PR and tackle that in another one, because that one is going to take some time no design and back and forth, and having these PRs staled is not necessary imo.

That we do. Yeah, agreed on not stalling all the PRs

File: src/tests/token/test_erc20.cairo [hunk 3]
```diff
@@ -38,10 +38,10 @@ fn test_initializer() {
...
     state.erc20.initializer(NAME, SYMBOL);
 
-    assert(state.erc20.name() == NAME, 'Name should be NAME');
-    assert(state.erc20.symbol() == SYMBOL, 'Symbol should be SYMBOL');
-    assert(state.erc20.decimals() == DECIMALS, 'Decimals should be 18');
-    assert(state.erc20.total_supply() == 0, 'Supply should eq 0');
+    assert(state.erc20.name() == NAME, 'Should be NAME');
+    assert(state.erc20.symbol() == SYMBOL, 'Should be SYMBOL');
+    assert(state.erc20.decimals() == DECIMALS, 'Should be DECIMALS');
+    assert(state.erc20.total_supply() == 0, 'Should equalual 0');
```
Comment (martriay) [inline src/tests/token/test_erc20.cairo:44]:
```suggestion
    assert(state.erc20.total_supply() == 0, 'Should equal 0');
```

File: src/tests/token/test_erc20.cairo [hunk 4]
```diff
@@ -147,7 +147,7 @@ fn test__approve_from_zero() {
 fn test__approve_to_zero() {
     let mut state = setup();
     testing::set_caller_address(OWNER());
-    state.erc20._approve(OWNER(), Zeroable::zero(), VALUE);
+    state.erc20._approve(OWNER(), ZERO(), VALUE);
```
Comment (martriay) [inline src/tests/token/test_erc20.cairo:150]: why do we have `ZERO()` instead of using `Zeroable::zero()`?

**Review by martriay — APPROVED** [review]

File: src/tests/token/test_erc20.cairo (hunk 4 above)
Comment (ericnordelo) [reply src/tests/token/test_erc20.cairo:150]: IMO shorter and easier to read and follow (is not the same to ctrl+f Zeroable::zero that is used for everything zeroable, than to look for the ZERO() constant)

Comment (martriay) [reply src/tests/token/test_erc20.cairo:150]: we could simply use `zero()` from `Zeroable` if it's a matte of length. I don't see much value in it and the fact that is used for everything zero is not a problem. Otoh I don't think it's that important, there's no problem with `ZERO()` either.

File: docs/modules/ROOT/pages/api/erc20.adoc [hunk 5]
```diff
@@ -471,3 +471,43 @@ See <<IERC20-Transfer,IERC20::Transfer>>.
...
+
+[.contract]
+[[ERC20]]
+=== `++ERC20++` link:https://github.com/OpenZeppelin/cairo-contracts/blob/release-v0.8.0-beta.0/src/presets/erc20.cairo[{github-icon},role=heading-link]
+
+```javascript
+use openzeppelin::presets::ERC20;
+```
+
+Basic ERC20 preset leveraging xref:#ERC20Component[ERC20Component] with a fixed-supply mechanism for token distribution.
```
[the remark this answers is not in the document]
Comment (andrew-fleming) [reply docs/modules/ROOT/pages/api/erc20.adoc:485]: `contract` works, but I don't love `fixed supply mechanism`. My reasoning behind the hyphen is that `fixed` and `supply` work together like a compound adjective to describe `mechanism`. IMO the hyphen makes it explicit to the reader that `fixed` describes `supply` and not `mechanism`

**Pull request merged**